### PR TITLE
Fix a few nits throughout the code.

### PR DESCRIPTION
**Pull request by markusthoemmes**
The statusProber thing is potentially kinda bad even, if it's used at all 😅

### CODE REVIEW

**Comment by jmprusi** [comment]
@markusthoemmes the statusProber is more or less copied from the current one in knative serving. What's the issue? How can we improve it? thanks.

**Comment by davidor** [comment]
@markusthoemmes regarding the prober, It would be great if we could reuse the Prober from knative/serving directly, but when we looked at it some time ago it was not possible. Not sure if that has changed.

File: pkg/generator/generator_test.go [hunk 1]
```diff
@@ -92,26 +92,27 @@ func TestTrafficSplits(t *testing.T) {
 	var tr tracker.Interface
 	// Create the Kubernetes services associated to the Knative services that
 	// appear in the ingress above
-	err := createServicesWithNames(
+	if err := createServicesWithNames(
```
Comment (davidor) [inline pkg/generator/generator_test.go]: What do you do with cases like this in knative/serving? Do you enforce a linter or some guidelines?
It'd would be good to use in this project the same ones as in knative/serving.

Comment (markusthoemmes) [reply pkg/generator/generator_test.go]: Unfortunately upstream doesn't use a set linter either. I mostly adjust things on a best-can-do basis.

**Pull request merged**